Sources_v3/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -136,6 +137,7 @@ class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
         parentMessageId: MessageId,
         attachments: [_ChatMessageAttachment<ExtraData>.Seed],
         showReplyInChannel: Bool,
+        quotedMessageId: MessageId?,
```

Comment:
Hmmm, I'm not sure about this. Also it seems we have too many parameters in the method, `cid` and `parentMessageId` shouldn't be there since they are the ones provided by the controller. 

I'd suggest having two separate methods for this with the following arguments:
```
func createNewThreadReply(
        text: String,
        command: String?,
        arguments: String?,
        attachments: [_ChatMessageAttachment<ExtraData>.Seed],
        showReplyInChannel: Bool,
        extraData: ExtraData.Message,
        completion: ((Result<MessageId, Error>) -> Void)? = nil
    ) 
     
func createNewQuoteReply(
        text: String,
        command: String?,
        arguments: String?,
        attachments: [_ChatMessageAttachment<ExtraData>.Seed],
        extraData: ExtraData.Message,
        completion: ((Result<MessageId, Error>) -> Void)? = nil
     ) 
```

What do you think?

Reply:
The thing is it's possible to send quoted thread reply.
That is why I decided that adjusting existing public API will be least confusing option.